code_search/README.md

```diff
@@ -0,0 +1,52 @@
+# Semantic Code Search
+
+End-to-End Semantic Code Search on Kubeflow
+
+## Prerequisites
+
+* Python 2.7 (with `pip`)
+* Python `virtualenv`
+
+**NOTE**: This project uses Google Cloud Dataflow which only supports Python 2.7.
+
+## Setup
+
+* Setup Python Virtual Environment
+```
+$ virtualenv venv
+$ source venv/bin/activate
+```
+
+* Install [`gcloud`](https://cloud.google.com/sdk/gcloud/) CLI
+
+* Setup Application Default Credentials 
+```
+$ gcloud auth application-default login
+```
+
+* Enable Dataflow via Command Line (or use the Google Cloud Console)
+```
+$ gcloud services enable dataflow.googleapis.com
+```
+
+* Build and install package
+```
+$ python setup.py build install
+```
+
+# Pipeline
+
+## Pre-process Python Source files from Github to Google BigQuery
+
+A sample execution looks like:
+
+```
+$ python bin/process_github_archive.py -i files/select_github_archive.sql -o code_search:function_docstrings -p kubeflow-dev -j bq2gs \
```

Comment:
Where is `process_github_archive.py`?

Reply:
Thank you for catching that. It looks like the top level `.gitignore` file has an ignore directive for `bin/` and never got committed. I'll have to rename the folder.